contrib/kubespray/script/openpai-generator.py

```diff
@@ -153,13 +154,51 @@ def get_kubernetes_node_info_from_API():
     return ret
 
 
+def get_pai_daemon_resource_request(cfg):
+    ret = {
+        "cpu-resource": 0,
+        "mem-resource": 0
+    }
+    if "qos-switch" not in cfg or cfg["qos-switch"] == "false":
```

Comment:
Have you checked all PAI services, including 3rd party daemons are all without resource request specified if the qos-switch is false?
![image](https://user-images.githubusercontent.com/32826762/91689317-6cba6c80-eb96-11ea-808c-44d7a68ec5f8.png)
![image](https://user-images.githubusercontent.com/32826762/91689329-7217b700-eb96-11ea-8dc7-f8398f3daac7.png)

Reply:
We starts to deploy PAI services after all third party services are ready. QoS switch doesn't affect the 3rd-party services.
And 3rd-party resource requests is calculated by: https://github.com/microsoft/pai/blob/364aee769a421771ab76284d55fd7039767d3f20/contrib/kubespray/script/openpai-generator.py#L145-L149

Do not need to recalculate here

Reply:
Are you sure all the 3rd-party pod resource are deducted from the node.status.allocatable?

Reply:
Sure,  3rd-party services already boot-up at this point. If the service not ready, installation will failed.

Reply:
3rd-party services allocated resource is in node allocatable, but not node free. See https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/

Reply:
Fxied